narwhals/utils.py

```diff
@@ -352,10 +356,14 @@ def to_native_namespace(self) -> ModuleType:
         Returns:
             Native module.
         """
-        if importer := _IMPORT_FUNCTIONS.get(self):
-            return importer()
-        msg = f"Not supported Implementation: {self}"  # pragma: no cover
-        raise AssertionError(msg)  # pragma: no cover
+        if self is Implementation.UNKNOWN:  # pragma: no cover
+            msg = f"Not supported Implementation: {self}"
+            raise AssertionError(msg)
+
+        importer = _IMPORT_FUNCTIONS.get(
+            self, lambda: import_optional_dependency(self.value, MIN_VERSIONS[self])
+        )
```

Comment:
The default is somehow a hack, yet avoids to bloat the `_IMPORT_FUNCTIONS` dict with same code